### DISCUSSION TITLE
Marker in lists

### DISCUSSION THREAD

**Discussion by SwitWu** [bdtc/lwarp#15] [Ideas]
Currently the lwarp package puts the list markers into the `li` tag together with the list contents and set inline style `list-style-type:none` to `ul`.

I think it's better that the markers can be handled separately like this [post answer](https://stackoverflow.com/a/23610216/18440497). In this way the list markers can have an indent (positive or negative), which makes the list clearer. A classic example could be found here: https://www.beamer.plus/Licenses-Copyright.html

**Comment by bdtc** [comment]
The list marker is generated by LaTeX, and it could be almost anything, and change at any time.

It may be possible to have Lwarp use a `<dl>` for itemize and enumerate so that the marker is separated from the item:
`<dl class="itemize">`, `<dl class="enumerate">`, `<dl class="description">`

This seems like a worthwhile change.

**Comment by bdtc** [comment]
Using a description list seems to be incorrect semantics.  For an itemize or enumerate, the list item does not describe the marker.

Instead, perhaps the following:

```
<ul class="enumerate">
<li> <span class="listmarker">1.</span>The first enumerated item.</li>
</ul>
```

Similar for itemize.

**Comment by SwitWu** [comment]
I'm wondering whether we can do something like this:

When lwarp sees `\begin{enumerate}[(a)]` or `\begin{enumerate}[label=(\alpha*)]`, it can produce an `ol` list with class `pap` (`p` for parenthesis, `a` for lowercase letter). Then we can use the power of CSS:
```CSS
@counter-style pap {
  system: extends lower-alpha;
  prefix: "(";
  suffix: ") ";
}
ol.pap {
  list-style: pap;
}
```

And here is a minimal file for my idea:

```html
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Document</title>
  <style>
    @counter-style pap {
      system: extends lower-alpha;
      prefix: "(";
      suffix: ") ";
    }
    @counter-style p1p {
      system: extends lower-decimal;
      prefix: "(";
      suffix: ") ";
    }
    ol.pap {
      list-style: pap;
    }
    ol.p1p {
      list-style: p1p;
    }
  </style>
</head>
<body>
  <ol class="pap">
    <li>
      <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Explicabo, voluptatem adipisci nesciunt excepturi culpa saepe eos accusantium tempore esse veniam consectetur rerum voluptate, tempora optio suscipit ea placeat? Optio, saepe?</p>
    </li>
    <li>
      <p>Lorem ipsum dolor sit amet consectetur, adipisicing elit. Voluptatibus hic laudantium quia magni unde neque similique cupiditate ut sunt nam ipsam iusto vero excepturi laborum asperiores modi ducimus, dolorem tempora.</p>
    </li>
  </ol>
  <ol class="p1p">
    <li>
      <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Explicabo, voluptatem adipisci nesciunt excepturi culpa saepe eos accusantium tempore esse veniam consectetur rerum voluptate, tempora optio suscipit ea placeat? Optio, saepe?</p>
    </li>
    <li>
      <p>Lorem ipsum dolor sit amet consectetur, adipisicing elit. Voluptatibus hic laudantium quia magni unde neque similique cupiditate ut sunt nam ipsam iusto vero excepturi laborum asperiores modi ducimus, dolorem tempora.</p>
    </li>
  </ol>
</body>
</html>
```

**Comment by bdtc** [comment]
Too many possibilities, sorry.  Hopefully classes for the markers will be sufficient.

**Comment by bdtc** [comment]
v0.915 now has CSS classes for LaTeX lists and item markers.
